core/src/main/java/com/spotify/metrics/core/InstrumentedExecutorService.java

```diff
@@ -0,0 +1,245 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * This class was initially copied and modified from the codahale metrics project.
+ *
+ * For the appropriate license (there was no header) see LICENSE.codahale.txt
+ *
+ * It was copied from the following tree:
+ * https://github.com/dropwizard/metrics/tree/6d1fff844b7fc8855b81bb42b7125bd84f3a3e7a
+ */
+
+package com.spotify.metrics.core;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * This class is an adaptation of Dropwizard's InstrumentedExecutorService that uses
+ * {@link com.spotify.metrics.core.SemanticMetricRegistry}.
+ *
+ * Just as Dropwizard's, this class wraps an {@link ExecutorService} and monitors the number of
+ * tasks submitted, running, completed and keeps a {@link Timer} to track task duration.
+ *
+ * It will register the metrics with the provided base metric ID and a "what" tag to identify the
+ * metric object, e.g: "submitted", "running", etc. If no basic metric ID is provided a base metric
+ * with tag "executor" set to a sequential "executor-N" value is used.
+ */
+public class InstrumentedExecutorService implements ExecutorService {
+    private static final AtomicLong NAME_COUNTER = new AtomicLong();
+
+    private final ExecutorService delegate;
+    private final Meter submitted;
+    private final Counter running;
+    private final Meter completed;
+    private final Timer idle;
+    private final Timer duration;
+
+    public InstrumentedExecutorService(ExecutorService delegate, SemanticMetricRegistry registry) {
+        this(delegate, registry,
+            new MetricId().tagged("executor", "executor-" + NAME_COUNTER.incrementAndGet()));
+    }
+
+    public InstrumentedExecutorService(
+        ExecutorService delegate,
+        SemanticMetricRegistry registry,
+        MetricId baseMetricId) {
+        this.delegate = delegate;
+        this.submitted = registry.meter(baseMetricId.tagged("what", "submitted"));
```

Comment:
Add a `"unit"` tag for all `counter` and `meter` types, in this case the unit would be `task`.

Reply:
Fixed in b4c743e